packages/typespec-azure-core/lib/foundations.tsp

```diff
@@ -285,10 +285,8 @@ op Operation<
   Traits extends TypeSpec.Reflection.Model = {},
   ErrorResponse = Azure.Core.Foundations.ErrorResponse
 >(
-  ...TraitProperties<
-    Traits & VersionParameterTrait<ApiVersionParameter>,
-    TraitLocation.ApiVersionParameter
-  >,
+  ...TraitProperties<VersionParameterTrait<ApiVersionParameter>, TraitLocation.ApiVersionParameter>,
+  ...TraitProperties<Traits, TraitLocation.Parameters>,
```

Comment:
Ok so this was done on purpose because the addition of the trait parameters is done earlier in most cases.